src/operators.jl

```diff
@@ -115,9 +115,19 @@ function union{T<:AbstractGraph}(g::T, h::T)
     return r
 end
 
+isunique(iter::Range) = true
+isunique(iter::Set) = true
+isunique(iter) = (length(unique(iter)) == length(iter))
+
 #@doc "filter g to include only the vertices present in iter which should not have duplicates
 #returns the subgraph of g induced by set(iter) along with the mapping from the old vertex names to the new vertex names" ->
-function inducedsubgraph{T<:AbstractGraph}(g::T, iter)
+function induced_subgraph{T<:AbstractGraph}(g::T, iter)
+    !isunique(iter) && error("Vertices in subgraph list must be unique")
```

Comment:
My strong preference is to throw an error instead of trying to guess what the right thing to do is. That is,
`induced_subgraph(g,[1,2,3,4,1])` is probably not what the user wants. In addition, the problem is with ordering - I'm not sure `unique()` preserves ordering, so we might get corruption in the forward mapping (which `1` is used?)